mlflow/pyfunc/dbconnect_artifact_cache.py

```diff
@@ -0,0 +1,135 @@
+import json
+import os
+import subprocess
+from mlflow.utils.databricks_utils import is_in_databricks_runtime
+from mlflow.utils.file_utils import get_or_create_tmp_dir
+
+
+_global_dbconnect_artifact_cache = None
```

Comment:
why do we need to add lots of global variables?

Reply:
we can make `_global_dbconnect_artifact_cache` to be `DBConnectArtifactCache`'s static member. if you like.

Reply:
there a some global variables in this PR but I think they are not "many". we can check them case by case

Reply:
I see, let's avoid using `global` if possible.